### PR TITLE
👷 📦 Postgres tests for Jenkins

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Add parallel build Step to Jenkins file that runs the tests against PostgreSQL
2. Update the Jenkinsfile to the new design specs employed by the ProcessEngine Runtime
3. Delete obsolete docker testfile
4. Only send a full testlog to slack, if at least one test has failed

PR: #76

## How can others test the changes?

Watch a build for the Consumer API. 
For example:
https://ci.process-engine.io/blue/organizations/jenkins/process-engine_node-lts%2Fconsumer_api_meta/detail/feature%2Fadd_postgres_tests_for_jenkins/3/pipeline

You'll see that the tests are now performed against Sqlite and PostgreSQL.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -60,62 +60,169 @@ def slack_send_testlog(testlog) {
...
+          agent any
+          options {
+            skipDefaultCheckout()
+          }
+          steps {
+            dir('_integration_tests') {
+              unstash('post_build');
+
+              script {
+
```
Comment (rrrene) [inline Jenkinsfile]: Wir nutzen in den Jenkinsfiles nie diese Leerzeile, die in manchen unserer JS-Files genutzt wird. Aus Konsistenzgründen sollten wir den Stil hier nicht weiter fragmentieren.

```suggestion
```

File: Jenkinsfile [hunk 2]
```diff
@@ -60,62 +60,169 @@ def slack_send_testlog(testlog) {
...
+            label 'macos'
+          }
+          options {
+            skipDefaultCheckout()
+          }
+          steps {
+            dir('_integration_tests') {
+              unstash('post_build');
+
+              script {
```
Comment (rrrene) [inline Jenkinsfile]:
```suggestion
```

File: Jenkinsfile [hunk 3]
```diff
@@ -60,62 +60,169 @@ def slack_send_testlog(testlog) {
...
+                def db_database_host_external_task = "process_engine__external_task_repository__host=${postgres_host}";
+                def db_database_host_process_model = "process_engine__process_model_repository__host=${postgres_host}";
+                def db_database_host_flow_node_instance = "process_engine__flow_node_instance_repository__host=${postgres_host}";
+
+                def db_environment_settings = "${db_database_host_correlation} ${db_database_host_external_task} ${db_database_host_process_model} ${db_database_host_flow_node_instance}";
+
+                def postgres_settings = "--env POSTGRES_USER=${postgres_username} --env POSTGRES_PASSWORD=${postgres_password} --env POSTGRES_DB=${postgres_database}";
+
+                docker.image('postgres:11').withRun("${postgres_settings}") { c ->
+
```
Comment (rrrene) [inline Jenkinsfile]:
```suggestion
```

File: Jenkinsfile [hunk 4]
```diff
@@ -60,62 +60,169 @@ def slack_send_testlog(testlog) {
...
+                  docker.image('postgres:11').inside("--link ${c.id}:${postgres_host}") {
+                    sh "while ! pg_isready --host ${postgres_host} --username ${postgres_username} --dbname ${postgres_database}; do sleep 1; done"
+                  };
+
+                  docker.image("node:${NODE_VERSION_NUMBER}").inside("--link ${c.id}:${postgres_host} --env PATH=$PATH:/$WORKSPACE/node_modules/.bin") {
+
```
Comment (rrrene) [inline Jenkinsfile]:
```suggestion
```

File: Jenkinsfile [hunk 5]
```diff
@@ -60,62 +60,169 @@ def slack_send_testlog(testlog) {
...
+                    def npm_test_command = "node ./node_modules/.bin/cross-env ${node_env_settings} ${db_environment_settings} ./node_modules/.bin/mocha -t 20000 test/**/*.js test/**/**/*.js";
+
+                    postgres_exit_code = sh(script: "${npm_test_command} --colors --reporter mocha-jenkins-reporter --exit > consumer_api_test_results_postgres.txt", returnStatus: true);
+
+                    postgres_testresults = sh(script: "cat consumer_api_test_results_postgres.txt", returnStdout: true).trim();
+                    junit 'consumer_api_test_results_postgres.xml'
+                  };
+
```
Comment (rrrene) [inline Jenkinsfile]:
```suggestion
```

**Comment by ElRaptorus** [comment]
@rrrene done

**Pull request merged**